### PR TITLE
Fix various issues

### PR DESCRIPTION
**Pull request by Julian702**
## Description

This pr fixes various issues.

### CODE REVIEW

File: backend/services/frontend/handler/frontend.go [hunk 1]
```diff
@@ -614,6 +722,22 @@ func (e *Frontend) ModifyDeckHandler(c *fiber.Ctx) error {
...
+	userID := helper.GetUserIDFromContext(c)
+	rspDeleteDeck, err := e.cardDeckService.DeleteDeck(c.Context(), &pbCommon.DeckRequest{
+		UserID: userID,
+		Deck: &pbCommon.Deck{
+			DeckID: c.Params("deckID"),
+		},
+	})
+	if err != nil {
+		return err
+	} else if !rspDeleteDeck.Success {
```
Comment (darmiel) [inline backend/services/frontend/handler/frontend.go:735]: remove `else`

File: frontend/components/deck/DeckList.tsx [hunk 2]
```diff
@@ -38,8 +38,10 @@ export default function DeckList({
 	className = "",
 }: Readonly<DeckListProps>) {
 	const filteredDecks = useMemo(() => {
-		const filteredDecks = decks?.filter((deck) =>
-			deck.deckName.toUpperCase().includes(filter)
+		const filteredDecks = decks?.filter(
+			(deck) =>
+				deck.deckName.toUpperCase().includes(filter) ||
+				deck.deckDescription.toUpperCase().includes(filter)
```
Comment (darmiel) [inline frontend/components/deck/DeckList.tsx:44]: without looking at the implementation, it's unclear the the filter should be in UPPERCASE. either transform filter here (preferred) or add proper typings (`Uppercase<string>`) to props

File: frontend/components/flashcard/Card.tsx [hunk 3]
```diff
@@ -79,15 +77,15 @@ export const Card = ({
...
+										? "cursor-pointer"
 										: "text-gray-200 hover:cursor-not-allowed"
 								}`}
 								size={20}
 								onClick={() => setDelete(editable)}
 							/>
 						)}
 						{/* <Edit2
-							className="hover:cursor-pointer"
+							className="cursor-pointer"
```
Comment (darmiel) [inline frontend/components/flashcard/Card.tsx:88]: this ded code should be removed

File: frontend/components/flashcard/Card.tsx [hunk 4]
```diff
@@ -106,11 +104,7 @@ export const Card = ({
 						inputFieldStyle="primary"
 						inputFieldSize="xs"
 						onKeyUp={(event) => {
-							if (event.key === "Enter") {
-								createCard()
-									.then((result) => {})
-									.catch((error) => {});
-							}
+							if (event.key === "Enter") createCard();
 						}}
```
Comment (darmiel) [inline frontend/components/flashcard/Card.tsx:108]:
```suggestion
						onKeyUp={(event) => event.key === "Enter" && createCard()}
```

File: frontend/components/form/InputField.tsx [hunk 5]
```diff
@@ -148,13 +149,7 @@ export const InputField = forwardRef(
...
 								onClickIcon?.(event);
 							}}
-							onKeyUp={(event) => {
-								if (event.key === "Enter") {
-									event.target.dispatchEvent(
-										new Event("click", { bubbles: true })
-									);
-								}
-							}}
+							onKeyUp={(event) => clickOnEnter(event)}
```
Comment (darmiel) [inline frontend/components/form/InputField.tsx:152]:
```suggestion
							onKeyUp={clickOnEnter}
```

File: frontend/components/layout/Section.tsx [hunk 6]
```diff
@@ -53,17 +54,9 @@ export const Section = ({
...
+				className={`font-extrabold ${onClick ? "cursor-pointer" : ""}`}
 				onClick={onClick}
-				onKeyUp={(event) => {
-					if (event.key === "Enter") {
-						event.target.dispatchEvent(
-							new Event("click", { bubbles: true })
-						);
-					}
-				}}
+				onKeyUp={(event) => clickOnEnter(event)}
```
Comment (darmiel) [inline frontend/components/layout/Section.tsx:59]:
```suggestion
				onKeyUp={clickOnEnter}
```

File: frontend/components/layout/Section.tsx [hunk 7]
```diff
@@ -53,17 +54,9 @@ export const Section = ({
 			<Text
 				textStyle="primary"
 				textSize="xs"
-				className={`font-extrabold ${
-					onClick ? "hover:cursor-pointer" : ""
-				}`}
+				className={`font-extrabold ${onClick ? "cursor-pointer" : ""}`}
```
Comment (darmiel) [inline frontend/components/layout/Section.tsx:57]: use clsx already

File: backend/api-documentation/decks/new deck.bru [hunk 8]
```diff
@@ -16,7 +16,9 @@ auth:bearer {
 
 body:json {
   {
-    "deckName": "test deck"
+    "deckName": "test deck",
+    "deckType": "PRIVATE",
+    "deckDescription": "description"
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply backend/api-documentation/decks/new deck.bru]: implemented in f572d3a

File: backend/services/frontend/handler/frontend.go [hunk 9]
```diff
@@ -347,73 +340,116 @@ func (e *Frontend) GetGroupMembersHandler(c *fiber.Ctx) error {
...
+	rspInviteUser, err := e.collaborationService.AddGroupUserInvite(c.Context(), &pbCommon.GroupInviteRequest{
+		UserID: userID,
+		Group: &pbCommon.Group{
+			GroupID: c.Params("groupID"),
+		},
+		InviteUserEmail: data["invitedUserEmail"],
+	})
+	if err != nil {
+		return err
+	} else if !rspInviteUser.Success {
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply backend/services/frontend/handler/frontend.go]: implemented in f572d3a

File: backend/services/frontend/handler/frontend.go [hunk 10]
```diff
@@ -347,73 +340,116 @@ func (e *Frontend) GetGroupMembersHandler(c *fiber.Ctx) error {
...
+	rspInviteUser, err := e.collaborationService.RemoveGroupUserInvite(c.Context(), &pbCommon.GroupInviteRequest{
 		UserID: userID,
 		Group: &pbCommon.Group{
 			GroupID: c.Params("groupID"),
 		},
+		InviteUserEmail: data["invitedUserEmail"],
 	})
 	if err != nil {
 		return err
+	} else if !rspInviteUser.Success {
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply backend/services/frontend/handler/frontend.go]: removed in f572d3a

File: backend/services/frontend/handler/frontend.go [hunk 11]
```diff
@@ -454,73 +490,143 @@ func (e *Frontend) RemoveUserGroupRequestHandler(c *fiber.Ctx) error {
...
 		return helper.NewMicroNotSuccessfulResponseErr(helper.FrontendServiceID)
 	}
 	return c.SendStatus(200)
 }
 
-func (e *Frontend) RemoveUserGroupInviteHandler(c *fiber.Ctx) error {
-	var data map[string]string
-	if err := c.BodyParser(&data); err != nil {
+func (e *Frontend) DeleteFavoriteDeckHandler(c *fiber.Ctx) error {
+	var deck = &pbCommon.Deck{}
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply backend/services/frontend/handler/frontend.go]: implemented in f572d3a

File: backend/services/frontend/handler/frontend.go [hunk 12]
```diff
@@ -454,73 +490,143 @@ func (e *Frontend) RemoveUserGroupRequestHandler(c *fiber.Ctx) error {
...
+	}
+	decks := converter.ConvertToTypeArray(rspGetActiveDecks.Decks, converter.ProtoDeckToFiberDeckConverter)
+
+	return c.JSON(converter.FiberGetGroupDecksResponseBody{
+		Decks: decks,
+	})
+}
+
+func (e *Frontend) DeleteActiveDeckHandler(c *fiber.Ctx) error {
+	var deck = &pbCommon.Deck{}
```
[the remark this answers is not in the document]
Comment (TomRomeo) [reply backend/services/frontend/handler/frontend.go]: implemented inf572d3a

File: backend/services/frontend/handler/frontend.go (hunk 1 above)
Comment (TomRomeo) [reply backend/services/frontend/handler/frontend.go]: removed in f572d3a

**Review by TomRomeo — APPROVED** [review]

File: frontend/components/deck/Deck.tsx [hunk 13]
```diff
@@ -70,16 +70,8 @@ export const Deck = ({
 
 	return (
 		<Link
-			className={`group rounded-lg shadow-lg transition-transform hover:cursor-pointer ${className}`}
+			className={`group cursor-pointer rounded-lg shadow-lg transition-transform ${className}`}
```
[the remark this answers is not in the document]
Comment (Julian702) [reply frontend/components/deck/Deck.tsx]: removed in 4068d02ab7d9f1f05f0a7889981f843e254fdc3a

File: frontend/components/deck/DeckList.tsx (hunk 2 above)
Comment (Julian702) [reply frontend/components/deck/DeckList.tsx:44]: added in 4068d02ab7d9f1f05f0a7889981f843e254fdc3a

File: frontend/components/flashcard/Card.tsx (hunk 3 above)
Comment (Julian702) [reply frontend/components/flashcard/Card.tsx:88]: will be reimplemented soon

File: frontend/components/flashcard/Card.tsx (hunk 4 above)
Comment (Julian702) [reply frontend/components/flashcard/Card.tsx]: fixed in 4068d02ab7d9f1f05f0a7889981f843e254fdc3a

File: frontend/components/form/InputField.tsx (hunk 5 above)
Comment (Julian702) [reply frontend/components/form/InputField.tsx]: fixed in 4068d02ab7d9f1f05f0a7889981f843e254fdc3a

File: frontend/components/layout/Section.tsx (hunk 6 above)
Comment (Julian702) [reply frontend/components/layout/Section.tsx]: fixed in 4068d02ab7d9f1f05f0a7889981f843e254fdc3a

File: frontend/components/layout/Section.tsx (hunk 7 above)
Comment (Julian702) [reply frontend/components/layout/Section.tsx]: na gut

**Review by darmiel — APPROVED** [review]

**Review by 4KevR — APPROVED** [review]
Looks great, I will approve this PR to not block further development (because the following comments will be out of scope for this fix PR).

I have some thoughts about changes in this PR which we could maybe address in the future development (I am curious to get to know your thoughts on my comments).

First of all, I don't like the change that the group role is propagated in the deck struct as a response. I can understand that it is required for a proper display of information and possible actions in the frontend when accessing a deck page, but I don't like the strict direct coupling of the collaboration and carddeck micro-service (also with regards to multiple calls to the collaboration service when `GetGroupDecks` is invoked). Also I think it is a bit misleading that it is called `deckRole` but is actually the `groupRole`. We should therefore discuss whether we want to be able to serve a role that is different from the group role (in my opinion out of scope at the moment).

I think this situation generally leads to the question on how we want to design the authorization process in our application. Based on my research, I would suggest that we change our current "standard" which describes to maintain the authorization of entity access in every micro-service (which handles the specific entities). As our role management get's more complex (as this case perfectly shows), I would suggest to externalize security concerns into a dedicated micro-service which handles the authorization of entity access. Common principles include [RBAC](https://de.wikipedia.org/wiki/Role_Based_Access_Control) or [ABAC](https://en.wikipedia.org/wiki/Attribute-based_access_control) (more granular control). This new security service would obtain the following two primary responsibilities:

1. Is the entity access (which is happening at the point of question) allowed? (security service acts as a policy decision point which is reached after an API-call is directed to a micro-service which is then using a call to the security service in a policy enforcement point)
2. Which actions are allowed for a requested entity? (information about permitted entity access (like `editAllowed`, `deleteAllowed`) is e.g. requested by the frontend-proxy when a GET-request happens and can be added to the response information obtained from the micro-service which serves the entity values)

This approach would enable us to centralize security decisions and move the responsibility away from the different micro-services. There are available open source solutions which we could use in the scope of our application. One "prominent" example would be [Casbin](https://casbin.org).

Apart from the security discussion, I would also suggest that we create an internal guideline on the naming of frontend properties. Because it seems to me that we are changing the handling of upper and lower case back and forth.

**Pull request merged**